src/MuTalk-Model/RandomOperatorMutantSelectionStrategy.class.st

```diff
@@ -23,129 +20,62 @@ RandomOperatorMutantSelectionStrategy class >> percent: aPercentage [
 		  (PercentRandomMutantSelectionStrategy new percentageOfMutants: aPercentage)
 ]
 
-{ #category : 'adding' }
-RandomOperatorMutantSelectionStrategy >> addMutantIn: newColl from: oldColl using: random and: aNumber [
-
-	randomMutantSelection addMutantIn: newColl from: oldColl using: random and: aNumber
-]
-
-{ #category : 'accessing-defaults' }
-RandomOperatorMutantSelectionStrategy >> defaultMutationGenerationStrategy [
-
-	^ randomMutantSelection defaultMutationGenerationStrategy
-]
-
-{ #category : 'accessing' }
-RandomOperatorMutantSelectionStrategy >> highLimitCondition: aSize [
-
-	^ randomMutantSelection highLimitCondition: aSize
-]
-
-{ #category : 'initialization' }
-RandomOperatorMutantSelectionStrategy >> initialize [
-
-	randomMutantSelection initialize
-]
-
-{ #category : 'accessing' }
-RandomOperatorMutantSelectionStrategy >> lowLimitCondition [
-
-	^ randomMutantSelection lowLimitCondition
-]
-
-{ #category : 'generating' }
-RandomOperatorMutantSelectionStrategy >> methodsToMutateFrom: aMutationTestingAnalysis [
-
-	^ randomMutantSelection methodsToMutateFrom: aMutationTestingAnalysis
-]
-
-{ #category : 'generating' }
-RandomOperatorMutantSelectionStrategy >> mutationsFor: aMutationTestingAnalysis [
-
-	^ randomMutantSelection mutationsFor: aMutationTestingAnalysis
-]
-
 { #category : 'generating' }
 RandomOperatorMutantSelectionStrategy >> mutationsFor: aMutationTestingAnalysis loggingIn: aLogger [
 
-	^ randomMutantSelection
-		  mutationsFor: aMutationTestingAnalysis
-		  loggingIn: aLogger
-]
-
-{ #category : 'generating' }
-RandomOperatorMutantSelectionStrategy >> mutationsFor: aMethod usingAll: aCollectionOfMutantOperators logginIn: aLogger [
-
-	^ randomMutantSelection
-		  mutationsFor: aMethod
-		  usingAll: aCollectionOfMutantOperators
-		  logginIn: aLogger
-]
-
-{ #category : 'accessing' }
-RandomOperatorMutantSelectionStrategy >> mutationsGenerationStrategy [
+	| mutations shuffledMutations mutationDict operators operator random |
+	mutations := mutationsGenerationStrategy
+		             mutationsFor: aMutationTestingAnalysis
+		             loggingIn: aLogger.
 
-	^ randomMutantSelection mutationsGenerationStrategy
-]
+	mutationDict := mutations groupedBy: [ :e | e operator ].
+	operators := mutationDict keys.
+	shuffledMutations := OrderedCollection new.
+	random := Random new.
 
-{ #category : 'accessing' }
-RandomOperatorMutantSelectionStrategy >> mutationsGenerationStrategy: aMutationsGenerationStrategy [
+	1 to: mutations size do: [ :i |
+		operator := self
+			            selectOperatorFrom: operators
+			            and: mutationDict
+			            using: random
+			            for: shuffledMutations.
 
-	^ randomMutantSelection mutationsGenerationStrategy:
-		  aMutationsGenerationStrategy
-]
+		shuffledMutations add: (self
+				 selectMutantFrom: mutationDict
+				 with: operator
+				 using: random
+				 for: shuffledMutations) ].
```

Comment:
Probably we can rewrite this in a more direct fashion using streams.
We can sit together later this week and check.
I wonder how this is tested (the code is below in the PR ^^)